### DISCUSSION TITLE
How to configure kiali with Is too canary révisions?

### DISCUSSION THREAD

**Discussion by yogeek** [kiali/kiali#7151] [Q&A]
I am using istio canary revisions (1.18 and 1.20 are present in the cluster at the same time, in the future it will be 1.20 and 1.22, etc) in order to allow users to choose when they want to upgrade to the latest revision.
How is it supposed to work with kiali please ? does kiali knows how to handle multiple istio revisions ?
Apparently, the `config_map_name` has to be specified in the CR but do I have to choose only 1 revision ?
(same thing with the `istio_sidecar_injector_config_map_name`)

I saw this https://github.com/kiali/kiali/issues/6042 but I do not understand if its purpose is to cover both revisions or only one ?

**Comment by jshaughn** [comment]
Hi @yogeek , I believe in the Canary scenario your Kiali will need to choose which Istio revision it will monitor.  See https://kiali.io/docs/installation/installation-guide/advanced-install-options/#multiple-istio-control-planes-in-the-same-cluster.

**Reply by yogeek** [reply]
@jshaughn thanks for the pointer, it is clear now 👍
